otter/test/test_controller.py

```diff
@@ -1464,3 +1471,96 @@ def test_checks_pass_replace_true_no_purge_failure(self):
         ])
         self.assertRaises(ValueError, self._remove, True, False,
                           seq_dispatcher)
+
+    def test_perform_convergence_remove_from_group(self):
+        """
+        Perform :func:`convergence_remove_server_from_group` with the given
+        dispatcher.
+        """
+        self.state.desired = 2
+        fake_super = FakeSupervisor()
+
+        dispatcher = ComposedDispatcher([
+            test_dispatcher(),
+            SequenceDispatcher([
```

Comment:
We also probably ought to save this sequence dispatcher in a local variable and use the `with ...consume():` pattern.